variable/include/scipp/variable/variable.h

```diff
@@ -93,28 +93,28 @@ class SCIPP_VARIABLE_EXPORT Variable {
     return variances<T>()[0];
   }
 
-  Variable slice(const Slice params) const;
+  [[nodiscard]] Variable slice(Slice params) const;
   void validateSlice(const Slice &s, const Variable &data) const;
-  [[maybe_unused]] Variable &setSlice(const Slice params, const Variable &data);
+  [[maybe_unused]] Variable &setSlice(Slice params, const Variable &data);
 
-  void rename(const Dim from, const Dim to);
+  void rename(Dim from, Dim to);
```

Comment:
What is wrong with passing these as const? We don't want to modify them.

Reply:
They are still const in the definition. But it is not necessary to use const in the declaration and clang-tidy warns about it.
This is a question of style, I think. But from what I understand, the general consensus is to avoid const in parameters that are passed by value because it is an implementation detail.

Reply:
I see, people seem to be getting into fights about this. While I agree that it is an implementation detail I feel it may be better to use the "const in both declaration and definition" style? Won't thinks gets messy otherwise if you change things from inline to not inline? It also makes it harder to add new functions unless have you have an IDE that does the "const" removal automatically.

Reply:
> Won't thinks gets messy otherwise if you change things from inline to not inline?

Does not really happen very often.

> It also makes it harder to add new functions unless have you have an IDE that does the "const" removal automatically.

I can't follow. Do you mean copy-pasting the function header between header and implementation files?

I can revert those changes. But then i would disable https://clang.llvm.org/extra/clang-tidy/checks/readability-avoid-const-params-in-decls.html because inspections are too noisy already.

Reply:
Leave it.